src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -424,10 +424,26 @@ public ObservableCollection<StyleItem> StyleItemsList
         public void AddStyle(StyleItem style)
         {
             StyleItemsList.Add(style);
-            preferenceSettings.GroupStyleItemsList.Add(new GroupStyleItem { HexColorString = style.HexColorString, Name = style.GroupName });
+            preferenceSettings.GroupStyleItemsList.Add(new GroupStyleItem { HexColorString = style.HexColorString, Name = style.Name, IsDefault = style.IsDefault });
             RaisePropertyChanged(nameof(StyleItemsList));
         }
 
+        /// <summary>
+        /// This method will add the Defaul GroupStyles to the Preferences panel
+        /// </summary>
+        private void AddDefaultStyles()
+        {
+            var defaultGroupStylesList = StyleItemsList.Where(style => style.IsDefault == true);
+            //Just in case the Default profiles have not been added then are added.
+            if(defaultGroupStylesList != null && defaultGroupStylesList.Count() == 0)
+            {
+                AddStyle(new StyleItem() { Name = Res.GroupStyleDefaultActions, HexColorString = Res.GroupStyleDefaultActionsColor, IsDefault = true });
+                AddStyle(new StyleItem() { Name = Res.GroupStyleDefaultInputs, HexColorString = Res.GroupStyleDefaultInputsColor, IsDefault = true });
+                AddStyle(new StyleItem() { Name = Res.GroupStyleDefaultOutputs, HexColorString = Res.GroupStyleDefaultOutputsColor, IsDefault = true });
+                AddStyle(new StyleItem() { Name = Res.GroupStyleDefaultReview, HexColorString = Res.GroupStyleDefaultReviewColor, IsDefault = true });
+            }           
+        }
```

Comment:
Can we move these to constructor of `PreferencesSetting` ?

Reply:
@QilongTang I moved the functionality of creating the Default Styles from PreferencesViewModel to the PreferenceSettings class, the call to PreferenceSettings.AddDefaultStyles() method cannot be done in the PreferenceSettings constructor due that the constructor is executed first and then the serialization process is done, so the default styles were duplicated after reopening Dynamo, that's why the call to AddDefaultStyles() need to be done in DynamoModel.cs after the serialization process was already done.
For more details check the commit: https://github.com/DynamoDS/Dynamo/pull/12667/commits/9c2fd6c62099fe97950d1b73a59270939d0e62f9
Let me know if you agree with this solution. Thanks

Reply:
@RobertGlobant20 Can you point me where is the serialization process?

Reply:
@QilongTang the serialization for loading the information from DynamoSettings.xml file is done in the method PrefererenceSettings.Load (see last image below), this method is called in DynamoModel constructor (CreateOrLoadPreferences method, see second image below).

Then this is the flow of the problem that I found when loading the default styles in the PreferenceSettings constructor 
1. Open Dynamo for the first time, then the Default GroupStyles are added (nothing is read from DynamoSettings.xml) .
2. Exit Dynamo then the 4 default GroupStyles are saved in DynamoSettings.xml.
3. Open Dynamo for the second time, 
   3.1. PreferenceSettings constructor is executed so the Default GroupStyles are added (we have just 4 styles until now).
   3.2 The PrefererenceSettings.Load() method is executed so the next line is executed **settings = 
         serializer.Deserialize(fs) as PreferenceSettings** so the 
         DynamoSettings.xml is read and the GroupsStyles in the xml file 
          are added to the current ones already existing, so we end up with 8 default styles
4. And so on, every time that we open Dynamo the styles are re-added
So the solution was adding the default styles after serialization is happening.
Let me know if it solves your question.

![image](https://user-images.githubusercontent.com/61755417/157730379-77da71ee-614c-4f9b-90ab-8ba3dc932bd7.png)
![image](https://user-images.githubusercontent.com/61755417/157730603-36111a76-6faf-4c62-a280-b0dad57f3c3c.png)
![image](https://user-images.githubusercontent.com/61755417/157732638-910b540e-7a35-4343-81d9-0dea9f6a3f5a.png)

Reply:
Thank you @RobertGlobant20 , when Dynamo deserialize the xml, I thought it would be overwritten instead of appending styles.. And it seems other settings in preferences are fine? I would love to figure out why.

Reply:
I reviewed it with @RobertGlobant20 together, since we cant implement `equals()` for this class, looks like what is covered in PR is the best way for now.

Reply:
@QilongTang fixed in the next commit: https://github.com/DynamoDS/Dynamo/pull/12667/commits/153c984c241ae7cec64bea81c335f9824dd4c3c8